nodejs/aws-serverless/queue.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import { iam, lambda, sqs } from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+import { createLambdaFunction, Handler } from "./function";
+import { EventSubscription } from "./subscription";
+
+export interface QueueEvent {
+    Records: QueueRecord[];
+}
+
+export interface QueueRecord {
+    "messageId": string;
+    "receiptHandle": string;
+    "body": string;
+    "attributes": {
+        "ApproximateReceiveCount": string;
+        "SentTimestamp": string;
+        "SenderId": string;
+        "ApproximateFirstReceiveTimestamp": string;
+    };
+    "messageAttributes": Record<string, any>;
+    "md5OfBody": string;
+    "eventSource": string;
+    "eventSourceARN": string;
+    "awsRegion": string;
+}
+
+export type QueueEventHandler = Handler<QueueEvent, void>;
+
+/**
+ * Arguments to control the sqs subscription.
+ */
+export type QueueSubscriptionArgs = {
+    /**
+     * From https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html.
+     *
+     * The largest number of records that AWS Lambda will retrieve. The maximum batch size supported
+     * by Amazon Simple Queue Service is up to 10 queue messages per batch. The default setting is
+     * 10.
+     */
+    batchSize?: number;
```

Comment:
Perhaps invert this - include the description first then a `See: ...` for the reference to primary source for this.

Reply:
Sure!